src/protocol/dp/distributions.rs

```diff
@@ -0,0 +1,200 @@
+use rand::{
+    distributions::{uniform::SampleUniform, Distribution, Uniform},
+    Rng,
+};
+
+use std::{
+    fmt::Debug,
+    ops::{Add, Div, Mul, Neg, Sub},
+};
+
+pub trait Float:
```

Comment:
I am on the fence on whether we need support something other than `f64`.

Reply:
I'm not.  `f64` is not just fine, but the only real choice for something like that.  Quad-precision floating point is inefficient and unnecessary and `f32` has too much quantization noise.